src/codeu/chat/server/Controller.java

```diff
@@ -26,150 +27,133 @@
+            model.add(user);
 
-      user = new User(id, name, creationTime, password);
-      model.add(user);
+            LOG.info(
+                    "newUser success (user.id=%s user.name=%s user.time=%s)",
+                    id,
+                    name,
+                    creationTime);
 
-      LOG.info(
-          "newUser success (user.id=%s user.name=%s user.time=%s)",
-          id,
-          name,
-          creationTime);
+        } else {
 
-    } else {
+            LOG.info(
+                    "newUser fail - id in use (user.id=%s user.name=%s user.time=%s)",
+                    id,
+                    name,
+                    creationTime);
+        }
 
-      LOG.info(
-          "newUser fail - id in use (user.id=%s user.name=%s user.time=%s)",
-          id,
-          name,
-          creationTime);
+        return user;
     }
 
-    return user;
-  }
+    @Override
+    public Conversation newConversation(Uuid id, String title, Uuid owner, Time creationTime) {
 
-  @Override
-  public Conversation newConversation(Uuid id, String title, Uuid owner, Time creationTime) {
+        Conversation conversation = null;
 
-    final User foundOwner = model.userById().first(owner);
+        if (isIdFree(id)) {
+            conversation = new Conversation(id, owner, creationTime, title);
+            model.add(conversation);
 
-    Conversation conversation = null;
+            LOG.info("Conversation added: " + conversation.id);
+        }
 
-    if (foundOwner != null && isIdFree(id)) {
-      conversation = new Conversation(id, owner, creationTime, title);
-      model.add(conversation);
-
-      LOG.info("Conversation added: " + conversation.id);
+        return conversation;
     }
 
-    return conversation;
-  }
-
-  private Uuid createId() {
+    private Uuid createId() {
 
-    Uuid candidate;
+        Uuid candidate;
+        for (candidate = uuidGenerator.make();
+             isIdInUse(candidate);
+             candidate = uuidGenerator.make()) {
 
-    for (candidate = uuidGenerator.make();
-         isIdInUse(candidate);
-         candidate = uuidGenerator.make()) {
+            // Assuming that "randomUuid" is actually well implemented, this
+            // loop should never be needed, but just incase make sure that the
+            // Uuid is not actually in use before returning it.
 
-     // Assuming that "randomUuid" is actually well implemented, this
-     // loop should never be needed, but just incase make sure that the
-     // Uuid is not actually in use before returning it.
+        }
 
+        return candidate;
     }
 
-    return candidate;
-  }
-
-  private boolean isIdInUse(Uuid id) {
-    return model.messageById().first(id) != null ||
-           model.conversationById().first(id) != null ||
-           model.userById().first(id) != null;
-  }
+    private boolean isIdInUse(Uuid id) {
+        String findID = "ID = " + SQLFormatter.sqlID(id);
```

Comment:
You are leaking the fact that the model uses SQL here as well. You should be passing the Uuid to the model methods and then doing the necessary conversions to a string in the model.